numpy/_core/tests/test_multiarray.py

```diff
@@ -8712,12 +8720,19 @@ class TestArrayAttributeDeletion:
     def test_multiarray_writable_attributes_deletion(self):
         # ticket #2046, should not seqfault, raise AttributeError
         a = np.ones(2)
-        attr = ['shape', 'strides', 'data', 'dtype', 'real', 'imag', 'flat']
+        attr = ['shape', 'data', 'dtype', 'real', 'imag', 'flat']
         with suppress_warnings() as sup:
             sup.filter(DeprecationWarning, "Assigning the 'data' attribute")
             for s in attr:
                 assert_raises(AttributeError, delattr, a, s)
 
+        attr = ['strides']
+        with suppress_warnings() as sup:
+            sup.filter(DeprecationWarning, "Assigning the 'data' attribute")
```

Comment:
N.B./nit: you should be fine with `catch_warnings` now.  Python fixed the bugs that made `sup` necessary.